paasta_tools/kubernetes_tools.py

```diff
@@ -936,14 +947,20 @@ def get_hacheck_sidecar_container(
             # and to not cause rolling updates on everything at once this is a config option for now
             if not system_paasta_config.get_hacheck_match_initial_delay():
                 initial_delay = 10
+            period_seconds = 10
             readiness_probe = V1Probe(
                 _exec=V1ExecAction(
-                    command=self.get_readiness_check_script(system_paasta_config)
+                    command=self.get_readiness_check_prefix(
+                        system_paasta_config=system_paasta_config,
+                        initial_delay=initial_delay,
+                        period_seconds=period_seconds,
+                    )
+                    + self.get_readiness_check_script(system_paasta_config)
                     + [str(self.get_container_port())]
                     + self.get_registrations()
                 ),
                 initial_delay_seconds=initial_delay,
-                period_seconds=10,
+                period_seconds=period_seconds,
             )
```

Comment:
Should we also explicitly set our V1Probe's timeout_seconds=period_seconds (or anything >1) here, to better mimic our pre-1.18 behavior and give our first few readiness probes a better chance to succeed?

Reply:
they should actually keep running regardless of the `timeout_seconds` value since we're still using dockershim - and i've noticed that once at least one of these succeeds, then future probes are way more likely to succeed/run faster

that said, we do probably still wanna figure out why these are slow at the start (which is probably 'cause we're giving this sidecar too little cpu)

Reply:
hmm I thought from the 1.20 note in https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes this implies the default timeout_seconds is now being used for the probe success, which is why readiness is failing more often (since it now needs to happen within the default 1s)?

Reply:
ahh, ok - i misunderstood you! umm, i think either way will more or less have the same effect: once one of these actually passes the ones after should also pass

that said, we might want to do it just so that if we're still in a funky state by the time we stop using dickershim we run into less surprises :p